uniffi_bindgen/src/bindings/gecko/templates/SharedHeaderTemplate.h

```diff
@@ -0,0 +1,795 @@
+// This file was autogenerated by some hot garbage in the `uniffi` crate.
+// Trust me, you don't want to mess with it!
+
+#ifndef mozilla_dom_{{ ci.namespace()|class_name_webidl }}_Shared
+#define mozilla_dom_{{ ci.namespace()|class_name_webidl }}_Shared
+
+#include <functional>
+
+#include "nsTArray.h"
+#include "prnetdb.h"
+
+#include "mozilla/Casting.h"
+#include "mozilla/CheckedInt.h"
+#include "mozilla/ErrorResult.h"
+#include "mozilla/Utf8.h"
+
+#include "mozilla/dom/BindingDeclarations.h"
+#include "mozilla/dom/Record.h"
+#include "mozilla/dom/{{ ci.namespace()|class_name_webidl }}Binding.h"
+
+extern "C" {
+
+struct RustBuffer {
+  int64_t mLen;
+  uint8_t* mData;
+};
+
+struct RustError {
+  int32_t mCode;
+  char* mMessage;
+};
+
+{% for func in ci.iter_ffi_function_definitions() -%}
+{%- match func.return_type() -%}
+{%- when Some with (type_) %}
+{{ type_|type_ffi }}
+{% when None %}
+void
+{%- endmatch %}
+{{ func.name() }}(
+    {%- for arg in func.arguments() %}
+    {{ arg.type_()|type_ffi }} {{ arg.name() -}}{%- if loop.last -%}{%- else -%},{%- endif -%}
+    {%- endfor %}
+    {%- if func.has_out_err() -%}{%- if func.arguments().len() > 0 %},{% endif %}
+    RustError* out_err
+    {%- endif %}
+);
+
+{% endfor -%}
+
+}  // extern "C"
+
+namespace mozilla {
+namespace dom {
+// namespace {{ ci.namespace() }} {
```

Comment:
It would be nice if we could namespace the UniFFI bindings, but Firefox’s binding layer specifically expects all interface implementations to live in `mozilla::dom`.